src/dbus_api/api/manager_3_5/methods.rs

```diff
@@ -0,0 +1,117 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+use std::path::Path;
+
+use dbus::{arg::Array, Message};
+use dbus_tree::{MTSync, MethodInfo, MethodResult};
+use futures::executor::block_on;
+
+use crate::{
+    dbus_api::{
+        blockdev::create_dbus_blockdev,
+        pool::create_dbus_pool,
+        types::{DbusErrorEnum, TData, OK_STRING},
+        util::{engine_to_dbus_err_tuple, get_next_arg, tuple_to_option},
+    },
+    engine::{CreateAction, EncryptionInfo, Engine, KeyDescription, Pool, PoolIdentifier},
+    stratis::StratisError,
+};
+
+type EncryptionParams = (Option<(bool, String)>, Option<(bool, (String, String))>);
+
+pub fn create_pool<E>(m: &MethodInfo<'_, MTSync<TData<E>>, TData<E>>) -> MethodResult
+where
+    E: 'static + Engine,
+{
+    let base_path = m.path.get_name();
+    let message: &Message = m.msg;
+    let mut iter = message.iter_init();
+
+    let name: &str = get_next_arg(&mut iter, 0)?;
+    let devs: Array<'_, &str, _> = get_next_arg(&mut iter, 2)?;
+    let (key_desc_tuple, clevis_tuple): EncryptionParams = (
+        Some(get_next_arg(&mut iter, 3)?),
+        Some(get_next_arg(&mut iter, 4)?),
+    );
```

Comment:
It looks like the arg count is off by 1.

Reply:
Fixed, thanks.